src/lib/service/learnerRecordAPI/courseRecord/models/courseRecord.ts

```diff
@@ -0,0 +1,96 @@
+import { plainToClass } from 'class-transformer'
+
+import { Module } from '../../../../model'
+import { Record, RecordState } from '../../models/record'
+import { ModuleRecord } from '../../moduleRecord/models/moduleRecord'
+
+export enum CourseRecordPreference {
+	Liked = 'LIKED',
+	Disliked = 'DISLIKED',
+}
+
+export class CourseRecordResponse {
+	courseRecords: CourseRecord[]
+
+	constructor(courseRecords: CourseRecord[]) {
+		this.courseRecords = courseRecords
+	}
+}
+
+export class CourseRecord extends Record {
+	courseTitle: string
+	modules: ModuleRecord[]
+	preference?: CourseRecordPreference
+	lastUpdated?: Date
+	courseDisplayState?: string
+	required: boolean
+
+	constructor(
+		courseId: string,
+		userId: string,
+		state: RecordState = RecordState.Null,
+		modules: ModuleRecord[] = [],
+		courseTitle: string,
+		required: boolean,
+		preference?: CourseRecordPreference,
+		lastUpdated?: Date
+	) {
+		super(courseId, userId, state)
+		this.courseTitle = courseTitle
+		this.preference = preference
+		this.required = required
+		if (lastUpdated) {
+			this.lastUpdated = new Date(lastUpdated)
+		}
+
+		this.fillRecords(modules)
+	}
+
+	public upsertModuleRecord(moduleRecordId: number, moduleRecord: ModuleRecord) {
+		let existingModIndex
+		for (let i = 0; i < this.modules.length; i++) {
+			const mr = this.modules[i]
+			if (mr.id === moduleRecordId) {
+				existingModIndex = i
+			}
+		}
+
+		if (existingModIndex) {
+			this.modules[existingModIndex] = moduleRecord
+		} else {
+			this.modules.push(moduleRecord)
+		}
+	}
+
+	public hasBeenAddedToLearningPlan() {
+		return this.isNull()
+	}
+
+	public hasBeenRemovedFromLearningPlan() {
+		return this.isArchived()
+	}
+
+	public getModuleRecord = (moduleId: string) => {
+		const records = this.modules.filter(m => m.moduleId === moduleId)
+		if (records.length > 0) {
+			return records[0]
+		} else {
+			return undefined
+		}
+	}
+
+	public areAllRelevantModulesComplete(modules: Module[]) {
```

Comment:
I think the function name should've been a statement rather than a question: `allRelevantModulesAreCompleted`.

Reply:
I actually think it's widely accepted boolean functions should indeed be questions (although like anything there's a bit of opinion/personal preference too) - IMO it makes it a bit more readable and syntactically 'correct' when being read as a question.